### PR TITLE
:sparkles: Implement validation for email and exceptions for invalid JWT token.

### PR DESCRIPTION
**Pull request by grata-salve**
:sparkles: Implement validation for possible empty email and exceptions for invalid JWT token.

### CODE REVIEW

**Comment by CalculusEnjoyer** [comment]
|File|Coverage [6.98%]|:x:|
|:-|:-:|:-:|
|[FacebookClient.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/caf29aea8f14bfe23a2b4500b6c58055e3dad66a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FFacebookClient.java)|8.11%|:x:|
|[InvalidJwtTokenException.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/caf29aea8f14bfe23a2b4500b6c58055e3dad66a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FInvalidJwtTokenException.java)|0%|:x:|
|[EmptyFacebookEmailException.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/caf29aea8f14bfe23a2b4500b6c58055e3dad66a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FEmptyFacebookEmailException.java)|0%|:x:|

|Total Project Coverage|48.87%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/services/FacebookClient.java [hunk 1]
```diff
@@ -36,7 +36,20 @@ public FacebookUser getUser(String accessToken) {
...
+    try {
+      facebookUser = restTemplate
+          .getForObject(facebookGraphApiBase + path, FacebookUser.class, variables);
+    } catch (Exception exception) {
+      String errorInfo = exception.getMessage();
+      throw new InvalidJwtTokenException(errorInfo.substring(
+          errorInfo.indexOf("message") + 10,
+          errorInfo.indexOf("type") - 3));
+    }
+    if ((facebookUser != null ? facebookUser.getEmail() : null) == null) {
```
Comment (Archasmiel) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/FacebookClient.java:50]: Interesting solution, but why it has ternary operator if you can make it like below or similar:
if (facebookUser == null || facebookUser.getEmail() == null) {

Comment (grata-salve) [reply src/main/java/com/intellias/intellistart/interviewplanning/services/FacebookClient.java:50]: Yes, I agree. Will refactor it in the next commit.

Comment (Archasmiel) [reply src/main/java/com/intellias/intellistart/interviewplanning/services/FacebookClient.java:50]: So, can you finish this PR faster, or there are some troubles?

**Comment by CalculusEnjoyer** [comment]
|File|Coverage [13.46%]|:green_apple:|
|:-|:-:|:-:|
|[FacebookService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FFacebookService.java)|21.62%|:green_apple:|
|[UserService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|18.58%|:green_apple:|
|[JwtTokenAuthenticationFilter.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fconfig%2FJwtTokenAuthenticationFilter.java)|10.62%|:green_apple:|
|[JwtTokenProvider.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FJwtTokenProvider.java)|9.43%|:x:|
|[FacebookClient.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FFacebookClient.java)|8.33%|:x:|
|[JwtAuthenticationResponse.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fpayload%2FJwtAuthenticationResponse.java)|0%|:x:|
|[FacebookLoginRequest.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fpayload%2FFacebookLoginRequest.java)|0%|:x:|
|[InvalidJwtTokenException.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FInvalidJwtTokenException.java)|0%|:x:|
|[EmptyFacebookEmailException.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/5cb5e619af781c31e5e292447ae55c5a517948f8/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FEmptyFacebookEmailException.java)|0%|:x:|

|Total Project Coverage|70.2%|:green_apple:|
|:-|:-:|:-:|

**Review by Archasmiel — APPROVED** [review]

**Pull request merged**